### PR TITLE
Resolve "Add support to stylus".

### LINKED ISSUE #2: Use babel-preset-env to reduce direct dev-dependencies
babel-preset-env takes care of all dependencies and plugins needed to compile code to a certain target.

https://github.com/babel/babel-preset-env

### LINKED ISSUE #3: Add support to stylus.

### PR DESCRIPTION
**Pull request by gneutzling**
This MR closes https://github.com/picter/npm-package-boilerplate/issues/3 and closes https://github.com/picter/npm-package-boilerplate/issues/2

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      },
+      {
+        test: /\.svg/,
+        loaders: ['svg-loader'],
+      },
+    ],
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      'process.env': {
```
Comment (marcuscaum) [inline webpack.config.js]: Do we need this env variable?

File: .babelrc [hunk 2]
```diff
@@ -1,8 +1,3 @@
 {
-  "plugins": [
-    "transform-class-properties",
-    "transform-object-assign",
-    "transform-object-rest-spread"
-  ],
-  "presets": ["es2015", "react"]
+  "presets": ["env", "react"]
```
Comment (lhrnd0) [inline .babelrc]: Is this env preset helpful even without configuration (like target browser and environments)?

**Comment by lhrnd0** [comment]
Just a single question. LGTM! :shipit:

File: .babelrc (hunk 2 above)
Comment (gneutzling) [reply .babelrc]: I followed the documentation https://github.com/babel/babel-preset-env#usage

But I am not sure if we should use other configs. We can improve that in the future, what do you think?

Comment (noxan) [reply .babelrc]: If I remember correctly they deprecated a lot of the old presets in favor of the `env` preset. I like to see it here :)

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    index: './src/index.jsx',
+  },
+  output: {
+    path: BUILD_DIR,
+    filename: '[name].js',
+    libraryTarget: 'umd',
+    umdNamedDefine: true,
+  },
+  resolve: {
+    extensions: ['.js', '.jsx', '.svg'],
```
Comment (noxan) [inline webpack.config.js]: Should we really include `svg` here?

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        test: /\.jsx?/,
+        loaders: ['babel-loader'],
+      },
+      {
+        test: /\.styl$/,
+        loaders: ['style-loader', 'css-loader', 'stylus-loader'],
+      },
+      {
+        test: /\.css/,
+        loaders: ['css-loader'],
```
Comment (noxan) [inline webpack.config.js]: Does css loader work without `style-loader`?

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      },
+      {
+        test: /\.svg/,
```
Comment (noxan) [inline webpack.config.js]: Does it lacks of a `$` ending character?

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      {
+        test: /\.jsx?/,
+        loaders: ['babel-loader'],
+      },
+      {
+        test: /\.styl$/,
+        loaders: ['style-loader', 'css-loader', 'stylus-loader'],
+      },
+      {
+        test: /\.css/,
```
Comment (noxan) [inline webpack.config.js]: Does it lacks of a `$` ending character?

**Review by noxan — CHANGES_REQUESTED** [review]

**Review by noxan — APPROVED** [review]

File: webpack.config.js (hunk 1 above)
Comment (gneutzling) [reply webpack.config.js]: Actually we don't need that. Good catch, thanks!

**Pull request merged**